js/src/setup-ads/ads-stepper/index.js

```diff
@@ -55,21 +62,38 @@ const AdsStepper = ( { formProps } ) => {
 	};
 
 	const handleSetupAccountsContinue = () => {
-		continueStep( '2' );
+		continueStep( hasGoogleAdsConnection ? '1' : '2' );
 	};
 
 	const handleCreateCampaignContinue = () => {
-		continueStep( '3' );
+		continueStep( hasGoogleAdsConnection ? '2' : '3' );
 	};
 
-	return (
-		// This Stepper with this class name
-		// should be refactored into separate shared component.
-		// It is also used in the Setup MC flow.
-		<Stepper
-			className="gla-setup-stepper"
-			currentStep={ step }
-			steps={ [
+	const getSteps = () => {
```

Comment:
@kt-12 Instead of adding the accounts step afterwards, L96, it's better if we remove it from the steps. It's easier.
```js
let steps = [
	{
		key: '1',
		label: __( 'Set up your accounts', 'google-listings-and-ads' ),
		content: (
			<SetupAccounts onContinue={ handleSetupAccountsContinue } />
		),
		onClick: handleStepClick,
	},
	{
		key: '2',
		label: __( 'Create your paid campaign', 'google-listings-and-ads' ),
		content: (
			<AdsCampaign
				trackingContext="setup-ads"
				onContinue={ handleCreateCampaignContinue }
			/>
		),
		onClick: handleStepClick,
	},
	{
		key: '3',
		label: __( 'Set up billing', 'google-listings-and-ads' ),
		content: <SetupBilling formProps={ formProps } />,
		onClick: handleStepClick,
	},
];

if ( hasGoogleAdsId ) {
	// Remove first step if there's an Ads account connected.
	steps.shift();

	steps = steps.map( ( singleStep ) => {
		return {
			...singleStep,
			key: ( parseInt( singleStep.key, 10 ) - 1 ).toString(),
		};
	}, [] );
}
```

then to update the `key` property of each step, you can use the `map` function.

Reply:
This works. Thanks.